include/dlr_allocator.h

```diff
@@ -0,0 +1,106 @@
+#ifndef DLR_ALLOCATOR_H_
+#define DLR_ALLOCATOR_H_
+
+#include <dmlc/common.h>
+#include <dmlc/logging.h>
```

Comment:
Looks like `dlr_allocator.h` only needs `<memory>` header.
In addition to it `dlr_allocator.cc` also needs `<stdlib.h>`